lib/routes/avatar/post.js

```diff
@@ -54,15 +57,19 @@ module.exports = {
       return reply(AppError.unsupportedProvider(payload.url));
     }
     var id = img.id();
-    db.addAvatar(id, uid, payload.url, provider, payload.selected)
-    .done(function() {
-      var info = {
-        event: 'avatar.post',
-        uid: uid
-      };
-      logger.info('activityEvent', info);
-      reply({ id: hex(id) }).code(201);
-    }, reply);
+
+    customs.checkAuthenticated('127.0.0.1', token, uid)
```

Comment:
Add the proper  ip address by adding: https://github.com/mozilla/fxa-auth-server/blob/10ee3224f271f48500b84c96786ce78c95c5a7f6/lib/server.js#L217 to fxa-profile-server server.js and use `request.app.clientAddress` (example https://github.com/mozilla/fxa-auth-server/blob/master/lib/routes/account.js#L368)